main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala

```diff
@@ -26,19 +26,31 @@ import org.json4s.JsonAST.JObject
 import org.json4s.JsonDSL._
 
 import scala.annotation.tailrec
+import ca.uwaterloo.flix.language.ast.TypedAst.Expr
+import ca.uwaterloo.flix.util.Result.ToErr
+import ca.uwaterloo.flix.language.ast.SourcePosition
+import ca.uwaterloo.flix.language.ast.shared.Source
+import ca.uwaterloo.flix.language.ast.TypedAst
 
 object HoverProvider {
-
   def processHover(uri: String, pos: Position)(implicit index: Index, root: Root, flix: Flix): JObject = {
-    index.query(uri, pos) match {
-      case None => mkNotFound(uri, pos)
+    hover(uri, pos)
+  }
+
 
-      case Some(entity) => hoverEntity(entity, uri, pos)
+  def hover(uri: String, pos: Position)(implicit root: Root, flix: Flix): JObject = {
```

Comment:
This probably belongs in the visitor.

Reply:
im not sure I understand why it would. It feels specific to the hover provider, whereas the content of visitors, at least in theory, should be more general. Although, I can see that it might be a little unnatural for the user of visitors to have to manually take the head of the list and such. But in theory, a user might want the entire back search trace

Reply:
if you think its cleaner, I can move it into visitors

Reply:
I think maybe my comment was misplaced. I was thinking the stack stuff might be reusable for other providers?

Reply:
true. While it should be seperate from the visit functions, it might make sense to put it might be nice to have it in the module. I think you're right, that other providers might be able to reuse it :)